tests/e2e/module_consistency_test.go

```diff
@@ -0,0 +1,95 @@
+package e2e_test
+
+import (
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/kyma-project/lifecycle-manager/api/shared"
+	"github.com/kyma-project/lifecycle-manager/api/v1beta2"
+
+	. "github.com/kyma-project/lifecycle-manager/pkg/testutils"
+)
+
+var _ = Describe("Module Keep Consistent After Deploy", Ordered, func() {
+	kyma := NewKymaWithSyncLabel("kyma-sample", "kcp-system", v1beta2.DefaultChannel,
+		v1beta2.SyncStrategyLocalSecret)
+	module := NewTemplateOperator(v1beta2.DefaultChannel)
+	moduleCR := NewTestModuleCR(remoteNamespace)
+
+	InitEmptyKymaBeforeAll(kyma)
+	CleanupKymaAfterAll(kyma)
+
+	Context("Given SKR Cluster", func() {
+		It("When Kyma Module is enabled on SKR Kyma CR", func() {
+			Eventually(EnableModule).
+				WithContext(ctx).
+				WithArguments(runtimeClient, defaultRemoteKymaName, remoteNamespace, module).
+				Should(Succeed())
+		})
+
+		It("Then Module Resources are deployed on SKR cluster", func() {
+			By("And Module CR exists")
+			Eventually(ModuleCRExists).
+				WithContext(ctx).
+				WithArguments(runtimeClient, moduleCR).
+				Should(Succeed())
+			By("And Module Operator Deployment is ready")
+			Eventually(DeploymentIsReady).
+				WithContext(ctx).
+				WithArguments(runtimeClient, "template-operator-controller-manager", TestModuleResourceNamespace).
+				Should(Succeed())
+
+			By("And KCP Kyma CR is in \"Ready\" State")
+			Eventually(KymaIsInState).
+				WithContext(ctx).
+				WithArguments(kyma.GetName(), kyma.GetNamespace(), controlPlaneClient, shared.StateReady).
+				Should(Succeed())
+		})
+		It("Then synced module resources keep consistent by verify resourceVersion not update", func() {
```

Comment:
I think this could be renamed to `And synced module resources remain consistent with the same resourceVersion`. And this should be a By in the previous `It` block

Reply:
I agree with the title, but put this in the previous `It` block, it affects the readability, this big truck of code is worth a dedicated block, and previous `It` block is checks in general this module deployed, this block check if they are not changed.

Reply:
If it's not placed in the previous It block, the e2e report will have incorrect indentation before this statement as far as I know. For the sake of this, we could extract the logic to check the resources version in a function and call the function here and then it would be more readable WDYT?